### PR TITLE
Add pull secret to deployment

### PR DESCRIPTION
**Pull request by athornton**

### CODE REVIEW

**Review by rra — COMMENTED** [review]
I'm confused about what this is trying to do. I can think of a couple of ways of adding pull secrets to each chart:

- Add a way to pass `imagePullSecrets` down from `values.yaml` and then add the necessary pull-secret glue to the environmental values files in lsp-deploy, not here, since that's where we're adding the pull-secret chart.
- Make each chart here, not in lsp-deploy, depend on the pull-secrets chart as a subchart, and then key off of whether the chart has enabled the pull-secrets subchart in its own values.yaml file or some override.

This seems to fall between those two stools, so I'm not sure how it could possibly work. There's no way for this chart to see whether the pull-secret chart is enabled, since it has no relationship to that chart.

**Comment by athornton** [comment]
That would probably be why the templating doesn't work.

File: charts/postgres/templates/deployment.yml [hunk 1]
```diff
@@ -18,6 +18,10 @@ spec:
         - name: {{ template "postgres.fullname" . }}
           imagePullPolicy: "Always"
           image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+          {{- with .Values.pull_secret }}
```
Comment (rra) [inline charts/postgres/templates/deployment.yml:21]: You want `if`, not `with`

**Review by rra — APPROVED** [review]
Okay with one typo.

**Pull request merged**